jung-algorithms/src/main/java/edu/uci/ics/jung/layout/algorithms/TreeLayoutAlgorithm.java

```diff
@@ -90,22 +91,27 @@ protected void buildTree(LayoutModel<N> layoutModel) {
     this.currentY = 20;
     Set<N> roots = TreeUtils.roots(layoutModel.getGraph());
     Preconditions.checkArgument(roots.size() > 0);
-    calculateDimensionX(layoutModel, roots);
+    int overallWidth = calculateDimensionX(layoutModel, roots);
+    overallWidth += (roots.size() + 1) * distX;
+    int overallHeight = calculateDimensionY(layoutModel, roots);
+    overallHeight += 2 * distY;
+    layoutModel.setSize(
+        Math.max(layoutModel.getWidth(), overallWidth),
+        Math.max(layoutModel.getHeight(), overallHeight));
     for (N node : roots) {
       calculateDimensionX(layoutModel, node);
-      double posX = this.basePositions.get(node) / 2 + this.distX;
-      buildTree(layoutModel, node, (int) posX);
+      currentX += (this.basePositions.get(node) / 2 + this.distX);
+      buildTree(layoutModel, node, (int) currentX);
     }
```

Comment:
changed to leverage the existing calculateDimensionX (and add a new calculateDimensionY) methods to set the layout size before doing any work.